### PR TITLE
Set custom user agent for Wikidata calls. Fixes #7383

### LINKED ISSUE #7383: Wikidata blocking OpenRefine because it doesn't set User-Agent string
When I try to upload some data to Wikidata from OpenRefine, my Wikidata log in details are not recognised. I message appears indicating 'invalid credentials', which is incorrect as I'm able to log in on a separate tab. 
### To Reproduce
1. First, do ... edits on my data, and reconciliations
2. Then, ... apply the schema, and click on upload to Wikidata
3. Finally, ... a pop up screen comes up asking me to enter the log in details, which doesn't recognise and indicates 'invalid credentials'

### Screenshots

Image

### Versions
 - Operating System:  WIndows and Mac
 - Browser Version:  newest versions of Chrome and Safari
 - JRE or JDK Version: 
 - OpenRefine:

### PR DESCRIPTION
**Pull request by tfmorris**
Fixes #7383

Changes proposed in this pull request:
- Set custom user agent on Wikidata Toolkit calls
- Set custom user agent on raw okhttp calls

### CODE REVIEW

**Comment by SoryRawyer** [comment]
@tfmorris what kind of manual testing do you have in mind for this? I'd like to help out if I can but I'm not very familiar with testing the Wikidata extension(s).

**Comment by SoryRawyer** [comment]
~Also, what do you think about running the snapshot release workflow on this branch (if possible) to make it easier for people on the forum test the changes?~

Edit: It doesn't look like we can manually run the snapshot release action without a change to the workflow definition. Since this is blocking an important usage of OpenRefine, is there a way to merge some functionality to `master` such that users can test a snapshot release?

**Comment by tfmorris** [comment]
The simplest test is just to go to Extensions -> Wikibase -> Manage Wikibase Account and attempt to login to Wikidata using your Wikidata consumer key. I've confirmed that this works for N=1 with my account, but a diversity of accounts would uncover any edge cases. If you (or anyone else) would like to test that part, that'd be great. After that, attempting to upload to Wikidata and Wikimedia Commons.

We can merge the PR after it's been reviewed and approved. Obviously, it's not going to get approved by reviewers without test coverage, so that's what I'm working on now.

**Comment by Abbe98** [comment]
I can confirm that it works with my (two) accounts.

**Comment by tfmorris** [comment]
I've removed the email address from the user agent string and added a (very basic) test. The code is structured in a way that doesn't make it super easy to test, but it's also not particularly high risk.

File: extensions/wikibase/src/org/openrefine/wikibase/utils/HttpClient.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class HttpClient {
+
+    public static final String USER_AGENT = "OpenRefine-Wikibase-extension/" + RefineServlet.FULL_VERSION
+            + " (https://openrefine.org)";
+
+    public static OkHttpClient getClient() {
+        OkHttpClient client = new OkHttpClient.Builder().addInterceptor(new UserAgentInterceptor(USER_AGENT + " okhttp/unknown")).build();
+        return client;
+    }
+
```
[the remark this answers is not in the document]
Comment (tfmorris) [reply extensions/wikibase/src/org/openrefine/wikibase/utils/HttpClient.java:21]: Not going to bother with this since it's super low priority.

**Review by SoryRawyer — APPROVED** [review]
I think the copilot suggestions seem reasonable to include but otherwise this all looks good to me. Thanks!

File: extensions/wikibase/src/org/openrefine/wikibase/commands/ConnectionManager.java [hunk 2]
```diff
@@ -221,7 +223,17 @@ public String getUsername(String mediaWikiApiEndpoint) {
...
 
-    private void setupConnection(ApiConnection connection) {
+    // Visible for testing only
+    static void setupConnection(ApiConnection connection) {
+        // Set our custom user agent (if it's not already set)
+        String defaultUserAgent = connection.getCustomUserAgent();
+        String agentLibrary = "";
+        if (defaultUserAgent != null && !defaultUserAgent.trim().isEmpty()) {
+            String[] parts = defaultUserAgent.trim().split(" ");
+            agentLibrary = parts.length > 0 ? parts[0] : "";
```
[the remark this answers is not in the document]
Comment (tfmorris) [reply extensions/wikibase/src/org/openrefine/wikibase/commands/ConnectionManager.java:233]: So amusing that it doesn't like its previous suggestion which was adopted verbatim -- plus the suggestion deletes a needed line.

[the remark this answers is not in the document]
Comment (SoryRawyer) [reply extensions/wikibase/src/org/openrefine/wikibase/commands/ConnectionManager.java:233]: Very odd. This read as somewhat harmless to me (except for the extra line removal) but I didn't realize it was effectively reverting its earlier suggestion.
